examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Inc/user_data.h

```diff
@@ -0,0 +1,39 @@
+#ifndef __USER_DATA_H__
+#define __USER_DATA_H__
+
+#define USER_DATA_STR_LEN 64 /* Must be a multiple of sizeof(int32_t) for CRC periph */
+
+typedef struct
+{
+    int32_t crc_checksum; /* For internal use only - Set by user_data_save_to_flash() */
+
+    int32_t wifi_client_encryption_mode; /* Casted from WiFi_Priv_Mode */
```

Comment:
Well, if we don't set the `wifi_` fields, they'll be saved as 0s and everything will work as expected. We'd be wasting a couple hundred bytes, and we'd have to explain in the Eth source code that some variables are not necessary, but we'd also save some implementation/maintenance trouble. Do you think it's worth the effort of maintaining 2 different user_data interfaces?

If we decide to remove them it won't be a big deal.